src/datasets/arrow_dataset.py

```diff
@@ -2080,16 +2133,17 @@ def _feature(values: Union[float, int, str, np.ndarray]) -> "tf.train.Feature":
                     raise ValueError(
                         f"values={values} is an np.ndarray with items of dtype {values[0].dtype}, which cannot be serialized"
                     )
-            if hasattr(values, "item"):  # single numpy values like np.int64 or np.str_
-                values = values.item()
-            if isinstance(values, float):
-                return _float_feature([values])
-            elif isinstance(values, int):
-                return _int64_feature([values])
-            elif isinstance(values, str):
-                return _bytes_feature([values.encode()])
+            if hasattr(values, "dtype"):
+                if np.issubdtype(values.dtype, np.float):
```

Comment:
This is probably gonna break somewhere, prefer the following:
```suggestion
                if np.issubdtype(values.dtype, np.floating):
```